server/schedule/checker/rule_checker_test.go

```diff
@@ -374,3 +384,116 @@ func (s *testRuleCheckerSuite) TestIssue3293(c *C) {
 	c.Assert(op, NotNil)
 	c.Assert(op.Desc(), Equals, "add-rule-peer")
 }
+
+func (s *testRuleCheckerSuite) TestIssue3299(c *C) {
+	s.cluster.AddLabelsStore(1, 1, map[string]string{"host": "host1"})
+	s.cluster.AddLabelsStore(2, 1, map[string]string{"dc": "sh"})
+	s.cluster.AddLeaderRegionWithRange(1, "", "", 1, 2)
+	s.ruleManager.SetStores(s.cluster.GetStores())
+
+	err := s.ruleManager.SetRule(&placement.Rule{
```

Comment:
How about making the test cases shorter since they have some common values in Rule?

Reply:
Done, that is the best I can do.